maintenance/monitor-component-metrics.md

```diff
@@ -3,43 +3,262 @@ title: Monitor Calico component metrics
+      - role: endpoints
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: felix-demo-svc
+        replacement: $1
+        action: keep
+    - job_name: 'typha_metrics'
+      scrape_interval: 5s
+      scheme: http
+      kubernetes_sd_configs:
+      - role: endpoints
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_service_name]
+        regex: typha-demo-svc
+        replacement: $1
+        action: keep
+EOF
+```
+
+## 3. Creating a service to expose Felix metrics
+
+Your prometheus configuration is going to check services and catch endpoints related to calico-demo-svc service, you need to create a service with that name to expose Felix metrics. This will help in discovery of your services and **makes** it dynamic. 
+
+> **Note**: Felix uses port 9091 TCP to publish its metrics.
+   {: .alert .alert-info}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: felix-demo-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-node
+  ports:
+  - port: 9091
+    targetPort: 9091
+EOF
+```
+
+## 4. Creating a service to expose Typha metrics
+
+You need to create a service that exposes your typha instance metrics to Prometheus.
+
+> **Note**: Typha uses **port 9091** TCP to publish it's metrics. However, in AWS Implementation of calico port is set to TCP Port 9093 via **TYPHA_PROMETHEUSMETRICSPORT** environment variable.
+   {: .alert .alert-warning}
+
+``` bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: typha-demo-svc
+  namespace: kube-system
+spec:
+  selector:
+    k8s-app: calico-typha
+  ports:
+  - port: 9093
+    targetPort: 9093
+EOF
+```
+
+## 5. Creating Prometheus pod
+
+Now that you allowed the user to view the metrics and have a valid config file for your Prometheus, it's time to create the pod.
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: prometheus-pod
+  labels:
+    app: prometheus-pod
+    role: monitoring
+spec:
+  containers:
+  - name: prometheus-pod
+    image: prom/prometheus
+    resources:
+      limits:
+        memory: "128Mi"
+        cpu: "500m"
+    volumeMounts:
+    - name: config-volume
+      mountPath: /etc/prometheus/prometheus.yml
+      subPath: prometheus.yml
+    ports:
+    - containerPort: 9090
+  volumes:
+  - name: config-volume
+    configMap:
+      name: promethous-config
+EOF
+```
+
+Check your cluster running pods to assure pod creation was successful.
+    
+    
+```bash
+kubectl get pods prometheus-pod
+```
+It should return something like the following.
+
+```
+NAME             READY   STATUS    RESTARTS   AGE
+prometheus-pod   1/1     Running   0          16s
+```
+
+You can access promethous dashboard by using port-forwarding feature.
+
+```bash
+kubectl port-forward pod/prometheus-pod 9090:9090
+```
+
+Browse to [http://localhost:9090](http://localhost:9090) you should be able to see Promethous dashboard.
```

Comment:
The dashboard is basically empty, adding some content about what you would expect to see and how to view the Felix metrics would be nice. How long do I have to wait? What do I need to to do to view the data?

Reply:
At step 5, my intention was how a user can access the prometheus dashboard since they have not enabled Felix yet it will be empty unless they go to step 6 and enable Felix monitoring and then create a simple Graph.

Maybe changing the sentence as below can help clarify this?

> you should be able to see an empty Prometheus dashboard.

Reply:
Ah, I was working from an older link to this document without these steps. Will follow through with these steps. Thanks.